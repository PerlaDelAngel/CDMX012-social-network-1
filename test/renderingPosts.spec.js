/**
 * @jest-environment jsdom
 */
import { createPosts } from '../src/components/renderingPosts.js';
/* import { genericModal } from '../src/components/genericModal.js'; */

jest.mock('../src/firebase-imports.js');

describe('Rendering posts:', () => {
  test('a new post is created and properly rendered', () => {
    const postData = {
      date: '01/04/2022',
      reading: 'Test title',
      text: 'Great book!',
      key: '123456789',
      likes: [],
      uid: 'userUid',
    };
    const name = 'Eli';
    const username = 'DinoEli';
    const currentUid = 'abc1234';

    document.body.innerHTML = '<div id="postsArea"></div>';
    createPosts(postData, currentUid, name, username);
    const postArea = document.querySelector('#postsArea');

    expect(postArea.innerHTML).toMatchSnapshot();
  });

  test('Renders like component of a post that the user has liked before', () => {
    const postData = {
      date: '01/04/2022',
      reading: 'Test title',
      text: 'Great book!',
      key: '123456789',
      likes: ['saraiG'],
      uid: 'userUid',
    };
    const name = 'Eli';
    const username = 'DinoEli';
    const currentUid = 'saraiG';

    document.body.innerHTML = '<div id="postsArea"></div>';
    createPosts(postData, currentUid, name, username);
    const postArea = document.querySelector('#postsArea');

    expect(postArea.innerHTML.includes('./assets/heart.png')).toBe(true);
  });

  test('Dropdown menu shows on click', () => {
    const postData = {
      date: '01/04/2022',
      reading: 'Test title',
      text: 'Great book!',
      key: '123456789',
      likes: [],
      uid: 'saraiG',
    };
    const name = 'Eli';
    const username = 'DinoEli';
    const currentUid = 'saraiG';

    document.body.innerHTML = '<div id="postsArea"></div>';
    createPosts(postData, currentUid, name, username);
    const optionsMenu = document.querySelector('.options-menu');
    optionsMenu.dispatchEvent(new Event('click'));

    const postArea = document.querySelector('#postsArea');

    expect(postArea.innerHTML).toMatchSnapshot();
  });

  test('Modal shows up after click on delete button', () => {
    const postData = {
      date: '01/04/2022',
      reading: 'Test title',
      text: 'Great book!',
      key: '123456789',
      likes: [],
      uid: 'saraiG',
    };
    const name = 'Eli';
    const username = 'DinoEli';
    const currentUid = 'saraiG';

    document.body.innerHTML = '<div id="postsArea"></div>';
    createPosts(postData, currentUid, name, username);
    const deleteOption = document.querySelector('.deletePost');
    deleteOption.dispatchEvent(new Event('click'));

    expect(document.body.innerHTML).toMatchSnapshot();
  });

  test('Edit function is executed properly on click', () => {
    const postData = {
      date: '01/04/2022',
      reading: 'Test title',
      text: 'Great book!',
      key: '123456789',
      likes: [],
      uid: 'saraiG',
    };
    const name = 'Eli';
    const username = 'DinoEli';
    const currentUid = 'saraiG';

    document.body.innerHTML = '<div id="postsArea"></div>';
    createPosts(postData, currentUid, name, username);
    const editOption = document.querySelector('.editPost');
    editOption.dispatchEvent(new Event('click'));

    expect(document.body.innerHTML).toMatchSnapshot();
  });
});
