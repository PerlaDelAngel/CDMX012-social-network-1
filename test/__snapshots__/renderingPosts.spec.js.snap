// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering posts: Dropdown menu shows on click 1`] = `
"<article class=\\"post-container\\"><div class=\\"info-user-post\\"><div class=\\"info\\"><p class=\\"nameProfile-P\\"><strong>Eli</strong></p><p class=\\"userName-P\\">@DinoEli</p><span class=\\"date-P\\">路 01/04/2022</span><img class=\\"options-menu\\" src=\\"./assets/options.png\\" height=\\"20\\"></div></div><div class=\\"to-edit\\"><div class=\\"post-content\\">
    <div><img src=\\"./assets/open-book.png\\" class=\\"book-icon\\"><p><strong>  Test title</strong></p></div> <br>
    <p>Great book!</p>
    <div></div></div></div><div class=\\"like-btn\\" id=\\"like-btn\\"><img src=\\"./assets/like.png\\" alt=\\"like empty button\\"><span class=\\"likeCount\\">0</span></div><div class=\\"dropdown-container show\\" tabindex=\\"-1\\"><div class=\\"dropdown\\"><a href=\\"#\\" class=\\"editPost\\" id=\\"editPost\\"><div>Edit</div></a><a href=\\"#\\" class=\\"deletePost\\" id=\\"deletePost\\"><div>Delete</div></a></div></div></article>"
`;

exports[`Rendering posts: Edit function is executed properly on click 1`] = `
"<div id=\\"postsArea\\"><article class=\\"post-container\\"><div class=\\"info-user-post\\"><div class=\\"info\\"><p class=\\"nameProfile-P\\"><strong>Eli</strong></p><p class=\\"userName-P\\">@DinoEli</p><span class=\\"date-P\\">路 01/04/2022</span><img class=\\"options-menu\\" src=\\"./assets/options.png\\" height=\\"20\\"></div></div><div class=\\"to-edit\\"><form class=\\"edit-form\\"><div class=\\"post-content\\">
  <div><img src=\\"./assets/open-book.png\\" class=\\"book-icon\\"> <input type=\\"text\\" class=\\"reading-txt\\" name=\\"reading\\" value=\\"Test title\\"></div>
  <br>
  <textarea name=\\"txt\\" class=\\"reading-description\\" rows=\\"5\\">Great book!</textarea>
  <br>
  </div></form><input type=\\"button\\" id=\\"btn-save\\" value=\\"Save\\"></div><div class=\\"like-btn\\" id=\\"like-btn\\"><img src=\\"./assets/like.png\\" alt=\\"like empty button\\"><span class=\\"likeCount\\">0</span></div><div class=\\"dropdown-container show\\" tabindex=\\"-1\\"><div class=\\"dropdown\\"><a href=\\"#\\" class=\\"editPost\\" id=\\"editPost\\"><div>Edit</div></a><a href=\\"#\\" class=\\"deletePost\\" id=\\"deletePost\\"><div>Delete</div></a></div></div></article></div>"
`;

exports[`Rendering posts: Modal shows up after click on delete button 1`] = `
"<div id=\\"postsArea\\"><article class=\\"post-container\\"><div class=\\"info-user-post\\"><div class=\\"info\\"><p class=\\"nameProfile-P\\"><strong>Eli</strong></p><p class=\\"userName-P\\">@DinoEli</p><span class=\\"date-P\\">路 01/04/2022</span><img class=\\"options-menu\\" src=\\"./assets/options.png\\" height=\\"20\\"></div></div><div class=\\"to-edit\\"><div class=\\"post-content\\">
    <div><img src=\\"./assets/open-book.png\\" class=\\"book-icon\\"><p><strong>  Test title</strong></p></div> <br>
    <p>Great book!</p>
    <div></div></div></div><div class=\\"like-btn\\" id=\\"like-btn\\"><img src=\\"./assets/like.png\\" alt=\\"like empty button\\"><span class=\\"likeCount\\">0</span></div><div class=\\"dropdown-container show\\" tabindex=\\"-1\\"><div class=\\"dropdown\\"><a href=\\"#\\" class=\\"editPost\\" id=\\"editPost\\"><div>Edit</div></a><a href=\\"#\\" class=\\"deletePost\\" id=\\"deletePost\\"><div>Delete</div></a></div></div></article></div><div class=\\"modal show-modal\\" id=\\"modal\\"><div class=\\"modal-content\\"><span class=\\"close-button\\" id=\\"close-button\\">x</span><h2>Are you sure you want to delete this post?</h2><button class=\\"accept-btn\\" id=\\"accept-btn\\">Yes</button></div></div>"
`;

exports[`Rendering posts: a new post is created and properly rendered 1`] = `
"<article class=\\"post-container\\"><div class=\\"info-user-post\\"><div class=\\"info\\"><p class=\\"nameProfile-P\\"><strong>Eli</strong></p><p class=\\"userName-P\\">@DinoEli</p><span class=\\"date-P\\">路 01/04/2022</span></div></div><div class=\\"to-edit\\"><div class=\\"post-content\\">
    <div><img src=\\"./assets/open-book.png\\" class=\\"book-icon\\"><p><strong>  Test title</strong></p></div> <br>
    <p>Great book!</p>
    <div></div></div></div><div class=\\"like-btn\\" id=\\"like-btn\\"><img src=\\"./assets/like.png\\" alt=\\"like empty button\\"><span class=\\"likeCount\\">0</span></div></article>"
`;
